src/AWS.Deploy.Common/Recipes/Validation/OptionSettingItemValidators/DockerBuildArgsValidator.cs

```diff
@@ -0,0 +1,41 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+using System;
+
+namespace AWS.Deploy.Common.Recipes.Validation
+{
+    /// <summary>
+    /// Validator for Docker build-time variables, passed via --build-arg
+    /// </summary>
+    public class DockerBuildArgsValidator : IOptionSettingItemValidator
```

Comment:
We should be support any command line argument passed into `docker build` and not just `--build-arg`. Basically make this like the `DotnetPublishArgsValidator` and just make sure they aren't passing in a docker file or tag.

Reply:
Fixed, and added a separate commit so we get a second entry in the release notes calling out the breaking change.

Here's how the next release look's so far via `git-chglog --next-tag TBD TBD`:
```
### Bug Fixes

* Rework the "Docker Build Args" option to support appending any options to the docker build command, instead of just `--build-arg`
* Validate and respect deployment bundle settings for the Docker or dotnet builds during server mode

### Build System or External Dependencies

* Version bump to 0.45

### Chore

* Add constructor injection support to option and recipe validators
* Add process exit code to improve telemetry of failed processes

### BREAKING CHANGE


If you were previously using "Docker Build Args" to set build-time variables, you must now prepend `--build-arg` for each argument yourself.

```